### DISCUSSION TITLE
Device Asset Tag Generation

### DISCUSSION THREAD

**Discussion by 0xab3d** [netbox-community/netbox#10840] [Ideas]
Hello,

Is there a plugin that the can generate the asset tag on the fly? For example, it takes the region's and site's slug and add a 6-digit number. 

Example:

Region: Net World Defender - slug:nwd
Site: London - slug: lon
Asset tag pattern -> nwdlon000001rtr

If not, would anyone be able to advise if the below is doable and shed a light on how to do it?
1. Assume an admin clicks on add new device.
2. Once an admin selects the Site, NetBox automatically adds the site's slug to the Asset tag field

**Comment by zombah** [comment]
Hello,
I don't know about plugins, but inventory systems like snipe-it or InvenTree will generate unique asset tags for you based on your tag choice and then you can create Custom Link for example which links your netbox device to inventory system.

**Reply by 0xab3d** [reply]
thank you, however this would introduce some duplication, no? I will have to define the inventory twice. We are trying to automate things and reduce manual effort as much as possible.
